node-src/tasks/report.ts

```diff
@@ -108,38 +128,56 @@ export const generateReport = async (ctx: Context) => {
     .property('storybookUrl', build.storybookUrl);
 
   for (const { status, result, spec, parameters, mode } of build.tests) {
-    const testSuffixName = mode.name || `[${parameters.viewport}px]`;
-    const suffix = parameters.viewportIsDefault ? '' : testSuffixName;
-    const testCase = suite
-      .testCase()
-      .className(spec.component.name.replaceAll(/[/|]/g, '.')) // transform story path to class path
-      .name(`${spec.name} ${suffix}`);
-
-    switch (status) {
-      case 'FAILED':
-        testCase.error('Server error while taking snapshot, please try again', status);
-        break;
-      case 'BROKEN':
-        testCase.error('Snapshot is broken due to an error in your Storybook', status);
-        break;
-      case 'DENIED':
-        testCase.failure('Snapshot was denied by a user', status);
-        break;
-      case 'PENDING':
-        testCase.failure('Snapshot contains visual changes and must be reviewed', status);
-        break;
-      default: {
-        if (['SKIPPED', 'PRESERVED'].includes(result)) {
-          testCase.skipped();
-        }
-      }
-    }
+    generateReportTestCase(suite, status, result, spec, parameters, mode);
```

Comment:
Do you think we should change this so we're passing in the entire test rather than destructuring just to pass in them in as positional arguments?

Reply:
Sure, I'll make that change.